server/src/main/java/io/crate/protocols/ssl/SslContextProvider.java

```diff
@@ -85,19 +98,65 @@ public SslContext getSslContext() {
 
     public void reloadSslContext() {
         synchronized (this) {
-            sslContext = buildSslContext();
+            sslContext = serverContext();
             LOGGER.info("SSL configuration is reloaded.");
         }
     }
 
-    private SslContext buildSslContext() {
-        var keystorePath = SslSettings.SSL_KEYSTORE_FILEPATH.get(settings);
-        var keystorePass = SslSettings.SSL_KEYSTORE_PASSWORD.get(settings).toCharArray();
-        var keystoreKeyPass = SslSettings.SSL_KEYSTORE_KEY_PASSWORD.get(settings).toCharArray();
+    public SSLContext jdkSSLContext() throws Exception {
```

Comment:
maybe add a comment that this is used for testing?

Reply:
Added a `VisibleForTesting` annotation